### DISCUSSION TITLE
pip installation

### DISCUSSION THREAD

**Discussion by hilaal** [marcomusy/vedo#356] [General]
Hi 
I am getting this error while installing with VCStudio or Jupyter.

I am using Python 3.7.1
Used command: pip3 install vedo
*****************************************
Collecting vedo
  Using cached vedo-2021.0.2.tar.gz (9.3 MB)
  Using cached vedo-2020.4.2.tar.gz (9.5 MB)
  Using cached vedo-2020.4.1.tar.gz (9.5 MB)
  Using cached vedo-2020.4.0.tar.gz (9.5 MB)
  Using cached vedo-2020.3.4.tar.gz (18.5 MB)
  Using cached vedo-2020.3.3.tar.gz (18.3 MB)
  Using cached vedo-2020.3.2.tar.gz (18.1 MB)
ERROR: Cannot install vedo==2020.3.2, vedo==2020.3.3, vedo==2020.3.4, vedo==2020.4.0, vedo==2020.4.1, vedo==2020.4.2 and vedo==2021.0.2 because these package versions have conflicting dependencies.

The conflict is caused by:
    vedo 2021.0.2 depends on vtk
    vedo 2020.4.2 depends on vtk
    vedo 2020.4.1 depends on vtk<9.0.0
    vedo 2020.4.0 depends on vtk
    vedo 2020.3.4 depends on vtk
    vedo 2020.3.3 depends on vtk
    vedo 2020.3.2 depends on vtk

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/user_guide/#fixing-conflicting-dependencies
Note: you may need to restart the kernel to use updated packages.

**Comment by marcomusy** [comment]
Hi @hilaal thanks for reporting, that's very strange!
Have you tried to uninstall first?
`pip3 uninstall vedo`
(repeat multiple times untill all versions disappear)

Currently we have in the requirements.txt this one line:
```
vtk==8.1.2,>=7
```
which seems ok by looking at the explanations from your link, i dont quite understand the error message..
any ideas?

**Comment by hilaal** [comment]
Hi 

Thank you ver much for your immediate response. I just followed your instructions. Other than that I don’t really understand anything :-). I had not installed Vedo previously. Despite of that I tried to uninstall it from my system. Here it is what I get…

WARNING: Skipping vedo as it is not installed.
hilaalalam@Hilaals-MacBook-Pro-4 ~ % pip3 uninstall vedo

WARNING: Skipping vedo as it is not installed.
hilaalalam@Hilaals-MacBook-Pro-4 ~ % pip3 install vedo 

Collecting vedo
  Using cached vedo-2021.0.2.tar.gz (9.3 MB)
  Using cached vedo-2020.4.2.tar.gz (9.5 MB)
  Using cached vedo-2020.4.1.tar.gz (9.5 MB)
  Using cached vedo-2020.4.0.tar.gz (9.5 MB)
  Using cached vedo-2020.3.4.tar.gz (18.5 MB)
  Using cached vedo-2020.3.3.tar.gz (18.3 MB)
  Using cached vedo-2020.3.2.tar.gz (18.1 MB)
ERROR: Cannot install vedo==2020.3.2, vedo==2020.3.3, vedo==2020.3.4, vedo==2020.4.0, vedo==2020.4.1, vedo==2020.4.2 and vedo==2021.0.2 because these package versions have conflicting dependencies.

The conflict is caused by:
    vedo 2021.0.2 depends on vtk
    vedo 2020.4.2 depends on vtk
    vedo 2020.4.1 depends on vtk<9.0.0
    vedo 2020.4.0 depends on vtk
    vedo 2020.3.4 depends on vtk
    vedo 2020.3.3 depends on vtk
    vedo 2020.3.2 depends on vtk

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/user_guide/#fixing-conflicting-dependencies


Regards
Hilaal ALAM

**Reply by marcomusy** [reply]
Try first this:
`pip3 uninstall vtk`
`pip3 install vedo -U`

If this (probably) doesn't work then try:
```bash
git clone https://github.com/marcomusy/vedo.git
cd vedo
rm requirements.txt
pip3 install .
```

**Comment by hilaal** [comment]
Hi Marco

Thanks. I have installed. The issue was with pip3. I upgraded the pip3 and then installed. Thank you for the support.

**Comment by hilaal** [comment]
Fine.. What I understood was...

**pip install vedo works alright as follows but pip3 install vedo does not work...**

hilaalalam@Hilaals-MacBook-Pro-4 ~ % pip install vedo
Defaulting to user installation because normal site-packages is not writeable
Requirement already satisfied: vedo in ./Library/Python/3.8/lib/python/site-packages (2021.0.2)
Requirement already satisfied: numpy in ./Library/Python/3.8/lib/python/site-packages (from vedo) (1.20.2)
Requirement already satisfied: vtk in ./Library/Python/3.8/lib/python/site-packages (from vedo) (9.0.1)

**pip3 install vedo does not work...**

hilaalalam@Hilaals-MacBook-Pro-4 ~ % pip3 install vedo
Collecting vedo
  Using cached vedo-2021.0.2.tar.gz (9.3 MB)
  Using cached vedo-2020.4.2.tar.gz (9.5 MB)
  Using cached vedo-2020.4.1.tar.gz (9.5 MB)
  Using cached vedo-2020.4.0.tar.gz (9.5 MB)
  Using cached vedo-2020.3.4.tar.gz (18.5 MB)
  Using cached vedo-2020.3.3.tar.gz (18.3 MB)
  Using cached vedo-2020.3.2.tar.gz (18.1 MB)
ERROR: Cannot install vedo==2020.3.2, vedo==2020.3.3, vedo==2020.3.4, vedo==2020.4.0, vedo==2020.4.1, vedo==2020.4.2 and vedo==2021.0.2 because these package versions have conflicting dependencies.

The conflict is caused by:
    vedo 2021.0.2 depends on vtk
    vedo 2020.4.2 depends on vtk
    vedo 2020.4.1 depends on vtk<9.0.0
    vedo 2020.4.0 depends on vtk
    vedo 2020.3.4 depends on vtk
    vedo 2020.3.3 depends on vtk
    vedo 2020.3.2 depends on vtk

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/user_guide/#fixing-conflicting-dependencies

**Reply by marcomusy** [reply]
thanks again for reporting this strange behavior ... indeed it seems the problem is with pip3 and vtk as
`pip3 install vtk` already gives an error on my mac, but not on my ubuntu system

**Comment by hilaal** [comment]
Yes. So any other way out? 

Because I am using Python3 for several projects and hence I cannot come out of that.

**Reply by marcomusy** [reply]
I can post a question to the VTK experts forum, but it may take a while to get an answer.

**Comment by hilaal** [comment]
Thank you very much. Let us wait.

**Comment by hilaal** [comment]
Hi 
I have moved to Ubuntu 20 on Mac. I have installed VEDO successfully. 
Python3 is at /usr/bin/python3
Vedo us at /usr/local/bin/vedo

********************************************************
Now I ran the following sample code
********************************************************
from vedo import Cone
c = Cone()
c.show(axes=1)
**********************************************************
In terminal Python3 it works fine. SUCCESS...!
************************************************************
But in **Visual Studio Code,** it throws the following error message

Traceback (most recent call last):
  File "vedo.py", line 1, in <module>
    from vedo import Cone
  File "/home/hilaal/Desktop/Simulation/vedo.py", line 1, in <module>
    from vedo import Cone
ImportError: cannot import name 'Cone' from partially initialized module 'vedo' (most likely due to a circular import) (/home/hilaal/Desktop/Simulation/vedo.py)
*****************************************************************

**Reply by marcomusy** [reply]
No! You cannot name a user script with the package name! Call it differently (not `vedo.py`)

**Reply by hilaal** [reply]
Oooooops... Yeaaahhh. How did I miss that? Thank you. It works... Sorry for bothering..

**Reply by marcomusy** [reply]
no worries :)
